tests/v2/test_waku_store_queue.nim

```diff
@@ -0,0 +1,269 @@
+    
+    check:
+      # Second page
+      pInfo.pageSize == 2
+      pInfo.direction == PagingDirection.BACKWARD
+      pInfo.cursor.senderTime == 1.0
+      err == HistoryResponseError.NONE
+      res.mapIt(it.timestamp.int) == @[1,2]
+    
+    (res, pInfo, err) = testStoreQueue.getPage(predicate,
+                                               pInfo)
+
+    check:
+      # Empty last page
+      pInfo.pageSize == 0
+      pInfo.direction == PagingDirection.BACKWARD
+      pInfo.cursor.senderTime == 1.0
+      err == HistoryResponseError.NONE
+      res.len == 0
+  
+  test "Store queue pagination works with predicate":
+    proc onlyEvenTimes(i: IndexedWakuMessage): bool = i.msg.timestamp.int64 mod 2 == 0
+    proc onlyOddTimes(i: IndexedWakuMessage): bool = i.msg.timestamp.int64 mod 2 != 0
+
+    ## Forward pagination: only even timestamped messages
+    
+    var (res, pInfo, err) = testStoreQueue.getPage(onlyEvenTimes,
+                                                   PagingInfo(pageSize: 2,
+                                                              direction: PagingDirection.FORWARD))
+    
+    check:
+      # First page
+      pInfo.pageSize == 2
+      pInfo.direction == PagingDirection.FORWARD
+      pInfo.cursor.senderTime == 4.0
+      err == HistoryResponseError.NONE
+      res.mapIt(it.timestamp.int) == @[2,4]
+    
+    (res, pInfo, err) = testStoreQueue.getPage(onlyEvenTimes,
+                                               pInfo)
+
+    check:
+      # Empty next page
+      pInfo.pageSize == 0
+      pInfo.direction == PagingDirection.FORWARD
+      pInfo.cursor.senderTime == 4.0
+      err == HistoryResponseError.NONE
+      res.len == 0
+
+    ## Backward pagination: only odd timestamped messages
+    
+    (res, pInfo, err) = testStoreQueue.getPage(onlyOddTimes,
+                                               PagingInfo(pageSize: 2,
+                                               direction: PagingDirection.BACKWARD))
+    
+    check:
+      # First page
+      pInfo.pageSize == 2
+      pInfo.direction == PagingDirection.BACKWARD
+      pInfo.cursor.senderTime == 3.0
+      err == HistoryResponseError.NONE
+      res.mapIt(it.timestamp.int) == @[3,5]
+    
+    (res, pInfo, err) = testStoreQueue.getPage(onlyOddTimes,
+                                               pInfo)
+
+    check:
+      # Next page
+      pInfo.pageSize == 1
+      pInfo.direction == PagingDirection.BACKWARD
+      pInfo.cursor.senderTime == 1.0
+      err == HistoryResponseError.NONE
+      res.mapIt(it.timestamp.int) == @[1]
+
+    (res, pInfo, err) = testStoreQueue.getPage(onlyOddTimes,
+                                               pInfo)
+
+    check:
+      # Empty last page
+      pInfo.pageSize == 0
+      pInfo.direction == PagingDirection.BACKWARD
+      pInfo.cursor.senderTime == 1.0
+      err == HistoryResponseError.NONE
+      res.len == 0
+
+  test "Store queue pagination handles invalid cursor":   
```

Comment:
How about having a similar test for `Forward` pagination (as forward and backward cases are handled by different procs)

Reply:
Good point! Thanks. Done in [a16c814](https://github.com/status-im/nim-waku/pull/849/commits/a16c814eb57921443115d41e199bb35eaf7ef546)